scalanet/src/io/iohk/scalanet/peergroup/StaticAddressMappedPeerGroup.scala

```diff
@@ -30,13 +30,13 @@ class StaticAddressMappedPeerGroup[A, AA, M](
     }
 
   override def server(): Observable[ServerEvent[A, M]] = {
-    underLyingPeerGroup
-      .server()
-      .collectChannelCreated
-      .map { underlyingChannel =>
+    underLyingPeerGroup.server().map {
+      case ChannelCreated(underlyingChannel) =>
         val a = reverseLookup(underlyingChannel.to)
         ChannelCreated(new ChannelImpl(a, underlyingChannel))
-      }
+      case HandshakeFailed(failure) =>
+        HandshakeFailed[A, M](new HandshakeException[A](reverseLookup(failure.to), failure.cause))
```

Comment:
I guess `reverseLookup` can fail here as well - in case of unknown node connecting - should we throw exception here? Maybe silently (or with logging) ignore the connection?